src/systems/filecoin_markets/storage_market/storage_market.id

```diff
@@ -0,0 +1,79 @@
+import addr "github.com/filecoin-project/specs/systems/filecoin_vm/actor/address"
+import block "github.com/filecoin-project/specs/systems/filecoin_blockchain/struct/block"
+import actor "github.com/filecoin-project/specs/systems/filecoin_vm/actor"
+import piece "github.com/filecoin-project/specs/systems/filecoin_files/piece"
+import deal "github.com/filecoin-project/specs/systems/filecoin_markets/storage_market/storage_deal"
+import ipld "github.com/filecoin-project/specs/libraries/ipld"
+
+type StorageDealProposalNotFound struct {}
+type StorageDealProposalRejected struct {}
+type StorageDealProposalAccepted struct {}
+type StorageDealProposalSigned struct {}
+type StorageDealPublished struct {}
+type StorageDealCommitted struct {}
+type StorageDealActive struct {}
+type StorageDealFailing struct {}
+type StorageDealRecovering struct {}
+type StorageDealExpired struct {}
+type StorageDealNotFound struct {}
+
+type StorageDealStatus union {
+    StorageDealProposalNotFound
+    StorageDealProposalRejected
+    StorageDealProposalAccepted
+    StorageDealProposalSigned
+    StorageDealPublished
+    StorageDealCommitted
+    StorageDealActive
+    StorageDealFailing
+    StorageDealRecovering
+    StorageDealExpired
+    StorageDealNotFound
+}
+
+type Signature struct {}
+
+type StorageAsk struct {
+    Price         actor.TokenAmount  // attoFIL per GiB per epoch
+    Collateral    actor.TokenAmount  // attoFIL per GiB per epoch
+
+    MinPieceSize  piece.PieceSize
+    Miner         addr.Address
+    Timestamp     block.ChainEpoch
+    MaxDuration   block.ChainEpoch
+    MinDuration   block.ChainEpoch
+    SeqNo         UVarint
+}
+
+type SignedStorageAsk struct {
+    Ask StorageAsk
+    Signature
+}
+
+type AskRequest struct {
+    Miner addr.Address
+}
+
+type AskResponse struct {
+    Ask SignedStorageAsk
+}
+
+type DealRequest struct {
+    DealProposal  deal.StorageDealProposal
+    Payload       ipld.CID  // root of the data to be transferred
+}
+
+type DealResponse struct {
+    State           StorageDealStatus
+
+    Message         string
+    Proposal        ipld.CID
```

Comment:
I don't know if @anorth thinks this is net net helpful (and I defer to him there) but spec convention has been to indicate a given object's CID by indicating &object. So in this case `ipld.CID` could be made more explicit as `&DealProposal` (see `block.id` for more examples)

Reply:
[blocking] looking at more, I think this would be helpful to do on all the `ipld.CID` inclusions you've added into the `id` files

Reply:
will do -- though a few CID's are really just that. like they Payload CID -- we don't know what type of data that's a CID for.

Reply:
not just a `storageDealProposal` CID?